src/Conversion/ONNXToTOSA/Math/Elementwise.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Elementwise.cpp - Elementwise Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX element-wise operators to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+#include "mlir/IR/TypeUtilities.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+template <typename ONNXOpT, typename TOSAOpT>
```

Comment:
This template works well and I am not asking you to change the code. 

You could still look at the template pattern we use in src/Conversion/Math/Elementwise.cpp where we use this:

```C++
template <>
struct ScalarOp<ONNXDivOp> {
  using FOp = arith::DivFOp;
  using IOp = arith::DivSIOp;
};
```
to reduce a bit on the redundant code that you have here. Let me know if you want to change it to that pattern. If not, I am fine accepting the PR as is with the understanding that if you add many more element wise op you would at least consider this pattern, if it fits your usage pattern.

Reply:
> @BraunPhilipp please let me know if you want to further change the code, or if you want the code to land as is.
> 
> In all cases, you need to confirm here that you abided by the DCO requirements. Once you do so, I can override the DCO failure when merging with main.

Sorry for the late reply due to Europe's vacation season. Thank you for the template suggestion! I will update it this week and double check the DCO requirements.

Reply:
From Swizerland myself... vacations are important :-)